tests/net/vlan/src/main.c

```diff
@@ -774,18 +774,21 @@ static void test_vlan_send_data(void)
 	net_context_unref(udp_v6_ctx);
 }
 
-void test_main(void)
+static void *setup(void)
 {
-	ztest_test_suite(net_vlan_test,
-			 ztest_unit_test(test_vlan_setup),
-			 ztest_unit_test(test_address_setup),
-			 ztest_unit_test(test_vlan_tci),
-			 ztest_unit_test(test_vlan_enable),
-			 ztest_unit_test(test_vlan_disable),
-			 ztest_unit_test(test_vlan_enable_all),
-			 ztest_unit_test(test_vlan_disable_all),
-			 ztest_unit_test(test_vlan_send_data)
-			 );
-
-	ztest_run_test_suite(net_vlan_test);
+	test_vlan_setup();
+	test_address_setup();
+	return NULL;
 }
+
+ZTEST(vlan_test_suite, test_vlan)
+{
+	test_vlan_tci();
+	test_vlan_enable();
+	test_vlan_disable();
+	test_vlan_enable_all();
+	test_vlan_disable_all();
+	test_vlan_send_data();
```

Comment:
Do all of them have a dependency problem? Could we define it by ZTEST macro as possible?

Reply:
> Do all of them have a dependency problem? Could we define it by ZTEST macro as possible?

enable/disable can not be split to different ZTEST. For others, they are dependent, and I have put them in different test functions.